pages/dao/[dao]/treasury/nfts/NFTs.tsx

```diff
@@ -81,20 +104,54 @@ const NFTs: NextPage<NFTsPageProps> = ({
       ) : (
         <>
           <div className={styles.header}>
-            <h1>All NFTs</h1>
-          </div>
-          <div className={styles.content}>
-            {nfts.map((nft, i) => (
-              // eslint-disable-next-line react/no-array-index-key
-              <div className={styles.card} key={`${i}`}>
-                <NFTCard
-                  name={nft.title}
-                  image={nft.uri}
-                  txHash={nft.transactionHash}
-                  contractId={nft.contractId}
+            <h1>NFTs</h1>
+            <div className={styles.selectors}>
+              <div className={styles.collections}>
+                <DropdownMultiSelect
+                  onChange={() => true}
+                  label="Collection"
+                  options={NFT_COLLECTIONS_DATA}
+                  simple
+                />
+              </div>
+              <div className={styles.contracts}>
+                <DropdownMultiSelect
+                  onChange={v => setCurrentContractIds(v)}
+                  defaultValue={currentContractIds}
+                  label="Smart contract"
+                  options={uniqueContracts}
+                  simple
                 />
               </div>
-            ))}
+            </div>
+          </div>
+          <div className={styles.content}>
+            {currentContractIds.length === 0 &&
+              nfts.map((nft, i) => (
+                // eslint-disable-next-line react/no-array-index-key
+                <div className={styles.card} key={`${i}`}>
+                  <NFTCard
+                    name={nft.title}
+                    image={nft.uri}
+                    txHash={nft.transactionHash}
+                    contractId={nft.contractId}
+                  />
+                </div>
+              ))}
+            {currentContractIds.length > 0 &&
+              nfts
+                .filter(ntf => currentContractIds.includes(ntf.contractId))
+                .map((nft, i) => (
+                  // eslint-disable-next-line react/no-array-index-key
+                  <div className={styles.card} key={`${i}`}>
```

Comment:
I consider there is no sense to use string template here. Maybe, transactionHash might work as a key?